src/clouds/body_of_clouds.rs

```diff
@@ -0,0 +1,317 @@
+use bevy::prelude::shape::{Box, Cube};
+use bevy::render::mesh::Indices;
+use bevy::render::pipeline::InputStepMode::Instance;
+use bevy::render::pipeline::PrimitiveTopology;
+use bevy::render::texture::{Extent3d, TextureDimension, TextureFormat};
+use bevy::render::{color::Color, shader::ShaderDefs, texture::Texture};
+use bevy::{
+    prelude::*,
+    reflect::TypeUuid,
+    render::{
+        mesh::shape,
+        pipeline::{PipelineDescriptor, RenderPipeline},
+        render_graph::{base, AssetRenderResourcesNode, RenderGraph},
+        renderer::RenderResources,
+        shader::{ShaderStage, ShaderStages},
+    },
+};
+use cgmath::num_traits::Float;
+use rand::prelude::*;
+use std::time::Instant;
+
+#[derive(RenderResources, Default, TypeUuid, ShaderDefs)]
+#[uuid = "69e20afc-b1c7-11eb-8529-0242ac130003"]
+pub(crate) struct MyMaterial {
+    #[shader_def]
+    pub color: Option<Handle<Texture>>,
+}
+
+const VERTEX_SHADER: &str = r#"
```

Comment:
I think we should put the shaders into separate files (1 file for vertex shader and 1 file for fragrment shader)